src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/Lock.cs

```diff
@@ -139,115 +194,135 @@ private bool TryAcquireContended(IntPtr currentThreadId, int millisecondsTimeout
             if (millisecondsTimeout == 0)
                 return false;
 
-            int spins = 1;
-
-            if (s_maxSpinCount == SpinningNotInitialized)
+            if (_spinLimit == SpinningNotInitialized)
             {
                 // Use RhGetProcessCpuCount directly to avoid Environment.ProcessorCount->ClassConstructorRunner->Lock->Environment.ProcessorCount cycle
-                s_maxSpinCount = (RuntimeImports.RhGetProcessCpuCount() > 1) ? MaxSpinningValue : SpinningDisabled;
+                s_processorCount = RuntimeImports.RhGetProcessCpuCount();
```

Comment:
```suggestion
                if (s_processorCount == 0)
                    s_processorCount = RuntimeImports.RhGetProcessCpuCount();
```
Avoid unnecessary writes of global state?

Reply:
It would make sense if `RhGetProcessCpuCount()` is expensive. I guess, we do not know, so we can do the check,

Reply:
The potentially most expensive part is cache-trashing caused by repeated writes.

Reply:
Ah, I see, we also do not want to overwrite and invalidate the shared state every time another lock initializes its spin count.